ampd/src/commands/register_public_key.rs

```diff
@@ -19,8 +20,18 @@ pub async fn run(config: Config, state_path: &Path) -> Result<Option<String>, Er
 
     let multisig_address = get_multisig_address(&config)?;
 
+    let tofnd_config = config.tofnd_config.clone();
+    let multisig_key = SharableEcdsaClient::new(
+        MultisigClient::connect(tofnd_config.party_uid, tofnd_config.url)
+            .await
+            .change_context(Error::Connection)?,
+    )
+    .keygen(&multisig_address.to_string())
+    .await
+    .change_context(Error::Tofnd)?;
+
```

Comment:
Can we add a log here? Something like "keygen successful for key_uid [multisig_address]"